docs/esi-leap-requirements.md

```diff
@@ -0,0 +1,15 @@
+# MOC esi-leap Requirements
+
+## Overview
+esi-leap is the leasing part of [Elastic Secure Infrastructure](https://github.com/CCI-MOC/esi) (esi), an OpenStack service for acting as a FLOCX provider to communicate with a FLOCX marketplace. It is intended to work on top of the [esi-common library](https://github.com/CCI-MOC/esi-common), which supports an OpenStack install that simulates Ironic multi-tenancy through the use of `project_owner_id` and `project_id` property attributes.
+
+## Definitions
+
+### Offer
+Baremetal node owners can offer up their nodes to others with nodes configurations, leasing period, and the price. On behalf of the node owners, esi-leap sends the offers to a FLOCX marketplace. An offer can be created, destroyed, saved, retrieved, and expired by esi-leap.
+
+### Contract
+A contract can allow nodes to be leased to a user within a specified period. esi-leap can create, destroy, save, retrieve, and expire a contract similar to an offer. Moreover, esi-leap is responsible for communicating with Ironic to fulfill the resources in the contracts.
+
+### Manager Service
+An esi-leap manager has periodic jobs to manage offers and contracts. Firstly, it retrieves and checks for new contracts from a FLOCX market. If new contracts are retrieved, it then fulfills the resources in the contracts and updates the status of the contracts to 'FULFILLED'. The second job is periodically expiring offers and contracts that are out-of-date. The manager will update these with an 'EXPIRED' status.
```

Comment:
I might suggest changing this paragraph into a list:

esi-leap manager has periodic jobs etc etc:

* expire offers
* fulfill contracts
* expire contracts

Lists are often an easier way for a reader to quickly parse meaning.